config/config.go

```diff
@@ -1,34 +1,48 @@
 package config
 
 import (
+	"fmt"
 	"io/ioutil"
-	"github.com/hashicorp/hcl"
-	"github.com/gruntwork-io/terragrunt/dynamodb"
-	"github.com/gruntwork-io/terragrunt/remote"
+
 	"github.com/gruntwork-io/terragrunt/errors"
+	"github.com/gruntwork-io/terragrunt/locks"
+	"github.com/gruntwork-io/terragrunt/remote"
+	"github.com/hashicorp/hcl"
 )
 
-const TERRAGRUNT_CONFIG_FILE = ".terragrunt"
+const configFilePath = ".terragrunt"
+
+// Config represents a parsed and expanded configuration
+type Config struct {
+	Lock        locks.Lock
+	RemoteState *remote.RemoteState
+}
 
-// A common interface with all fields that could be in the .terragrunt config file.
-type TerragruntConfig struct {
-	DynamoDbLock *dynamodb.DynamoDbLock
-	RemoteState  *remote.RemoteState
+// fileConfig represents the configuration supported in the .terragrunt file
+type fileConfig struct {
+	Lock        *LockConfig `json:"lock,omitempty"`
+	RemoteState *remote.RemoteState
+}
+
+// LockConfig represents generic configuration for Lock providers
+type LockConfig struct {
+	Backend string            `json:"backend"`
+	Config  map[string]string `json:"config"`
 }
 
 // Read the Terragrunt config file from its default location
-func ReadTerragruntConfig() (*TerragruntConfig, error) {
-	return parseTerragruntConfigFile(TERRAGRUNT_CONFIG_FILE)
+func Read() (*Config, error) {
```

Comment:
Same comment here about longer names. To a random user, `config.Read()` says very little about what this code does and forces you to dig into the code & docs to understand; `config.ReadTerragruntConfig()` tells you a whole lot more. With auto-complete in modern IDEs, the extra characters don't cost much :)